solidity/contracts/token/libs/FungibleTokenRouter.sol

```diff
@@ -2,16 +2,60 @@
 pragma solidity >=0.8.0;
 
 import {TokenRouter} from "./TokenRouter.sol";
+import {Quote} from "../../interfaces/ITokenBridge.sol";
 
 /**
  * @title Hyperlane Fungible Token Router that extends TokenRouter with scaling logic for fungible tokens with different decimals.
  * @author Abacus Works
  */
 abstract contract FungibleTokenRouter is TokenRouter {
-    uint256 public immutable scale;
+    uint128 public immutable scaleNumerator;
+    uint128 public immutable scaleDenominator;
+
+    uint128 public immutable feeNumerator;
+    uint128 public immutable feeDenominator;
+    address public immutable feeRecipient;
 
     constructor(uint256 _scale, address _mailbox) TokenRouter(_mailbox) {
-        scale = _scale;
+        scaleNumerator = uint128(_scale);
```

Comment:
So maybe this is overkill, but I'm wondering if we can make this a bit more general and configurable? I.e. make it a call to an external address where the router just sends the destination, recipient, amount parameters, and the call returns the quote? That way we can be more flexible and support flow based things in the future as well?

Reply:
The problem I see with fee "hooks" is token approvals.
Would prefer to have the client implementation simply be
```
Quotes[] quotes = warpRoute.quoteTransferRemote(.., .., amount);
for (quote in quotes) {
  quote.token.approve(warpRoute, quote.amount)
}
warpRoute.transferRemote{nativeQuote.amount}(.., ..,. amount);
```

Reply:
I see that the fee hook can simply compute the quote but the warp route can ultimately be the one that charges the fee.